### PR TITLE
RavenDB-7069

### PR DESCRIPTION
**Pull request by talweiss1982**
* added a logger to Async/AsyncServerClient.cs
* now fetching topology from the client in a sync manor to insure we have selected the correct request executor
* change client profiling to reflect on the underlining LRU implementation
* added force topology update for replication informer
* changed cluster topology cache and replication cache to have the same format
*  removed index replication test that is no longer valid, it was trying to prevent it from the client which is not supported, i made it deprecated.

### CODE REVIEW

**Comment by ayende** [comment]
@talweiss1982 tests are failing

File: Raven.Client.Lightweight/Connection/ReplicationInformerLocalCache.cs [hunk 1]
```diff
@@ -123,7 +123,7 @@ public static void ClearLocalCache(string serverHash)
 #endif
         }
 
-        public static List<OperationMetadata> TryLoadClusterNodesFromLocalCache(string serverHash)
+/*        public static List<OperationMetadata> TryLoadClusterNodesFromLocalCache(string serverHash)
```
Comment (ayende) [inline Raven.Client.Lightweight/Connection/ReplicationInformerLocalCache.cs]: Why is this commented (and not deleted)?

Comment (talweiss1982) [reply Raven.Client.Lightweight/Connection/ReplicationInformerLocalCache.cs]: i forgot to delete it.

**Comment by talweiss1982** [comment]
I'll go over the failing tests don't modify the PR so we won't run into conflicts

**Pull request merged**